tsdb/index/tsi1/partition_test.go

```diff
@@ -82,6 +83,73 @@ func TestPartition_Manifest(t *testing.T) {
 	})
 }
 
+var badManifestPath string = filepath.Join(os.DevNull, tsi1.ManifestFileName)
+
+func TestPartition_Manifest_Write_Fail(t *testing.T) {
+	const expectedError = "not a directory"
+	t.Run("write MANIFEST", func(t *testing.T) {
+		m := tsi1.NewManifest(badManifestPath)
+		_, err := m.Write()
+		if !strings.Contains(err.Error(), expectedError) {
+			t.Fatalf("expected: %q, got: %q", expectedError, err.Error())
+		}
+	})
+}
+
+func TestPartition_PrependLogFile_Write_Fail(t *testing.T) {
+	const expectedError = "not a directory"
+	t.Run("write MANIFEST", func(t *testing.T) {
+		sfile := MustOpenSeriesFile()
+		defer sfile.Close()
+
+		p := MustOpenPartition(sfile.SeriesFile)
+		defer func() {
+			if err := p.Close(); err != nil {
+				t.Fatalf("error closing partition: %v", err)
+			}
+		}()
+		p.Partition.MaxLogFileSize = -1
+		fileN := p.FileN()
+		p.CheckLogFile()
+		if fileN >= p.FileN() {
+			t.Fatalf("manifest write succeeded: expected more than %d files, got %d files", fileN, p.FileN())
+		}
+		p.SetManifestPathForTest(badManifestPath)
+		fileN = p.FileN()
+		p.CheckLogFile()
+		if fileN != p.FileN() {
+			t.Fatalf("manifest write failed: expected %d files, got %d files", fileN, p.FileN())
+		}
+	})
+}
+
+func TestPartition_Compact_Write_Fail(t *testing.T) {
+	const expectedError = "not a directory"
+	t.Run("write MANIFEST", func(t *testing.T) {
+		sfile := MustOpenSeriesFile()
+		defer sfile.Close()
+
+		p := MustOpenPartition(sfile.SeriesFile)
+		defer func() {
+			if err := p.Close(); err != nil {
+				t.Fatalf("error closing partition: %v", err)
+			}
+		}()
+		p.Partition.MaxLogFileSize = -1
+		fileN := p.FileN()
+		p.Compact()
+		if (1 + fileN) != p.FileN() {
+			t.Fatalf("manifest write succeeded: expected more than %d files, got %d files", fileN, p.FileN())
+		}
+		p.SetManifestPathForTest(badManifestPath)
+		fileN = p.FileN()
+		p.Compact()
+		if fileN != p.FileN() {
+			t.Fatalf("manifest write failed: expected %d files, got %d files", fileN, p.FileN())
```

Comment:
```suggestion
			t.Fatalf("manifest write should have failed the compaction, but number of files changed: expected %d files, got %d files", fileN, p.FileN())
```

Something like the suggestion might be more clear as error messages if these do fail, also elsewhere in the tests.

Reply:
```
// or with testify/assert:
assert.Equal(t, fileN, p.FileN(), "bad manifest write should have failed the compaction"); 
```